test/ApplicationConfigInjectionTest.php

```diff
@@ -0,0 +1,413 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Expressive;
+
+use Closure;
+use PHPUnit_Framework_TestCase as TestCase;
+use ReflectionFunction;
+use ReflectionProperty;
+use SplQueue;
+use Zend\Expressive\Application;
+use Zend\Expressive\Container\ApplicationFactory;
+use Zend\Expressive\ErrorMiddlewarePipe;
+use Zend\Expressive\Router\RouterInterface;
+use Zend\Stratigility\ErrorMiddlewareInterface;
+use Zend\Stratigility\MiddlewarePipe;
+
+/**
+ * Tests the functionality present in the ApplicationConfigInjectionTrait.
+ */
+class ApplicationConfigInjectionTest extends TestCase
+{
+    use ContainerTrait;
+
+    public function setUp()
+    {
+        $this->container = $this->mockContainerInterface();
+        $this->router = $this->prophesize(RouterInterface::class);
+    }
+
+    public function createApplication()
+    {
+        return new Application($this->router->reveal(), $this->container->reveal());
+    }
+
+    public function assertRoute($spec, array $routes)
+    {
+        $this->assertTrue(array_reduce($routes, function ($found, $route) use ($spec) {
+            if ($found) {
+                return $found;
+            }
+
+            if ($route->getPath() !== $spec['path']) {
+                return false;
+            }
+
+            if ($route->getMiddleware() !== $spec['middleware']) {
+                return false;
+            }
+
+            if (isset($spec['allowed_methods'])) {
+                if ($route->getAllowedMethods() !== $spec['allowed_methods']) {
+                    return false;
+                }
+            }
+
+            if (! isset($spec['allowed_methods'])) {
+                if ($route->getAllowedMethods() !== Route::HTTP_METHOD_ANY) {
+                    return false;
+                }
+            }
+
+            return true;
+        }, false));
+    }
+
+    public function callableMiddlewares()
+    {
+        return [
+           ['HelloWorld'],
+           [
+                function () {
+                }
+           ],
+           [[InvokableMiddleware::class, 'staticallyCallableMiddleware']],
+        ];
+    }
+
+    /**
+     * @dataProvider callableMiddlewares
+     */
+    public function testFactorySetsUpRoutesFromConfig($middleware)
+    {
+        $config = [
+            'routes' => [
+                [
+                    'path' => '/',
+                    'middleware' => $middleware,
+                    'allowed_methods' => [ 'GET' ],
```

Comment:
We shouldn't have spaces after opening array brackets and before closing:
`['GET']` instead of `[ 'GET' ]`

More places in this file.